experimental/Schemes/src/Resolution_tools.jl

```diff
@@ -0,0 +1,377 @@
+########################################################################
+# intersection matrix of exceptional divisors -- surface case
+########################################################################
+@doc raw"""
+    function intersection_matrix(phi::Union{BlowUpSequence,MixedBlowUpSequence})
+
+Given a desingularization `phi` of a surface (2-dimensional reduced scheme), 
+return the intersection matrix of the exceptional divisor of phi.
+
+Return a tuple `M`, `v`, `M2`, `m` where 
+`M` is the intersection matrix computed over the underlying field `k`, 
+`v` is the list of k-irreducible components of the exceptional divisor ordered as in the columns of `M`, 
+`M2` is the intersection matrix over the algebraic closure `kbar` of `k`,
+`m` the number of kbar-irreducible components of each entry of `v`.
+
+!!! warning
+  This is only applicable, if the `phi` is the desingularization of a 2-dimensional reduced scheme, as the exceptional divisor of the desingularization of higher dimensional schemes is not a curve.
+
+!!! note
+  The intersection matrix referred to in textbooks is `M2`, as these usually restrict to the case of algebraically closed fields, but computations are usually performed over suitable subfields, e.g. `QQ` instead of `CC`. 
+
+# Example
+```jldoctest
+julia> R,(x,y,z) = polynomial_ring(QQ,3);
+
+julia> W = AffineScheme(R);
```

Comment:
Shouldn't this be
```suggestion
julia> W = affine_scheme(R);
```
(also in the tests)